horreum-backend/src/main/java/io/hyperfoil/tools/horreum/svc/ActionServiceImpl.java

```diff
@@ -307,57 +308,42 @@ public void onNewExperimentResult(ExperimentService.ExperimentResult result) {
       executeActions(MessageBusChannels.EXPERIMENT_RESULT_NEW, result.profile.testId, result, result.notify);
    }
 
-   JsonNode exportTest(int testId) {
-      ArrayNode actions = JsonNodeFactory.instance.arrayNode();
-      for (ActionDAO action : ActionDAO.<ActionDAO>list("testId", testId)) {
-         ObjectNode node = Util.OBJECT_MAPPER.valueToTree(ActionMapper.from(action));
+   void exportTest(TestExport test) {
+      List<Action> actions = new ArrayList<>();
+      for (ActionDAO action : ActionDAO.<ActionDAO>list("testId", test.id)) {
+         Action a = ActionMapper.from(action);
          if (!action.secrets.isEmpty()) {
             try {
-               node.put("secrets", encryptionManager.encrypt(action.secrets.toString()));
+               a.secrets =  Util.OBJECT_MAPPER.valueToTree( encryptionManager.encrypt(action.secrets.toString()));
             } catch (GeneralSecurityException e) {
                throw ServiceException.serverError("Cannot encrypt secrets for action " + action.id);
             }
          }
-         actions.add(node);
+         actions.add(a);
       }
-      return actions;
+      test.actions = actions;
    }
 
-   void importTest(int testId, JsonNode actions, boolean forceUseTestId) {
-      if (actions.isMissingNode() || actions.isNull()) {
-         log.debugf("Import test %d: no actions");
-      } else if (actions.isArray()) {
-         log.debugf("Importing %d actions for test %d", actions.size(), testId);
-         for (JsonNode node : actions) {
-            if (!node.isObject()) {
-               throw ServiceException.badRequest("Test actions must be an array of objects");
+   void importTest(TestExport test) {
+      for (Action a : test.actions) {
+         try {
+            ActionDAO action = ActionMapper.to(a);
+            if (a.secrets != null) {
+               action.secrets = Util.OBJECT_MAPPER.readTree(encryptionManager.decrypt(a.secrets.textValue()));
+            } else {
```

Comment:
encryptionManager uses `horreum.db.secret` as the password for encryption. if this is different between Horreum instances, the secrets encryption/decryption will not work

Reply:
So, we need to think about/if we should serialize secrets, and provide a mechanism for redefining secrets in the new test

Reply:
Hm, it's decrypted when we export. Isn't that good enough?
We'll encrypt it when we import.

Reply:
On only support secrets when we update Tests?

Reply:
> Hm, we could also export some sort of "id" of the Horreum instance... If it doesn't match we set the secret to null?

How do users know if there secrets have been imported or not?  If we set it to null, we need to inform them that they need to configure the secrets again.

Reply:
That could be in a pop-up box, or it could be in the docs, but users need to know the limitations, how to check and what to do